pkg/scale/decode.go

```diff
@@ -0,0 +1,539 @@
+package scale
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/big"
+	"reflect"
+)
+
+func Unmarshal(data []byte, dst interface{}) (err error) {
```

Comment:
it would also be nice to support unmarshalling from an `io.Reader`

Reply:
That usually follows the `json.Decoder` interface if you want to supply an `io.Reader`. I'll create an improvement issue for this if you think it'll be worthwhile.  Right now all the data is buffered in memory.